git_test.go

```diff
@@ -1,35 +1,476 @@
+
+	currentBranch, err := r.repo.Head()
+	if err != nil {
+		r.t.Errorf(">> could not get head branch for fake repo: %v", err)
+	}
+
+	master, err := r.repo.Branch("master")
+	if err != nil {
+		r.t.Errorf(">> could not get master branch for fake repo: %v", err)
+	}
+
+	scm.baseRef = master.Name
+	scm.headSha = currentBranch.Hash().String()
+	scm.changeRequest = r.divergesFromBase
+
+	return scm
+}
+
+func TestCheckGitProvider(t *testing.T) {
+	// Prepare Github
+	originalSha := os.Getenv("GITHUB_SHA")
+	originalBaseRef := os.Getenv("GITHUB_BASE_REF")
+	originalHeadRef := os.Getenv("GITHUB_HEAD_REF")
+	cleanGithubFn := func() {
+		os.Unsetenv("GITHUB_SHA")
+		os.Unsetenv("GITHUB_BASE_REF")
+		os.Unsetenv("GITHUB_HEAD_REF")
+	}
+	restoreGithubFn := func() {
+		os.Setenv("GITHUB_SHA", originalSha)
+		os.Setenv("GITHUB_BASE_REF", originalBaseRef)
+		os.Setenv("GITHUB_HEAD_REF", originalHeadRef)
+	}
+
+	// prepare Gitlab
+	originalCommitBranch := os.Getenv("CI_COMMIT_BRANCH")
+	originalSourceBranchSha := os.Getenv("CI_MERGE_REQUEST_SOURCE_BRANCH_SHA")
+	originalTargetBranchName := os.Getenv("CI_MERGE_REQUEST_TARGET_BRANCH_NAME")
+	restoreGitlabFn := func() {
+		os.Setenv("CI_COMMIT_BRANCH", originalCommitBranch)
+		os.Setenv("CI_MERGE_REQUEST_SOURCE_BRANCH_SHA", originalSourceBranchSha)
+		os.Setenv("CI_MERGE_REQUEST_TARGET_BRANCH_NAME", originalTargetBranchName)
+	}
+
+	t.Run("Github", func(t *testing.T) {
+		if originalCommitBranch != "" {
+			t.Skip("Tests skipped when running on Gitlab")
+		}
+
+		testChangeRequest := false
+
+		testSha := "0123456"
+		if originalSha != "" {
+			testSha = originalSha
+		}
+
+		testBaseRef := ""
+		if originalBaseRef != "" {
+			testBaseRef = originalBaseRef
+		}
+
+		testHeadRef := "feature/pr-23"
+		if originalHeadRef != "" {
+			testHeadRef = originalHeadRef
+		}
+
+		t.Run("Running for Branches", func(t *testing.T) {
+			os.Setenv("GITHUB_SHA", testSha)
+			if originalBaseRef != "" || originalHeadRef != "" {
+				testChangeRequest = true
+			}
+			defer restoreGithubFn()
+
+			sha, baseRef, provider, changeRequest := checkGitProvider()
+			assert.Equal(t, testSha, sha)
+			assert.Equal(t, testBaseRef, baseRef)
+			assert.Equal(t, "Github", provider)
+			assert.Equal(t, testChangeRequest, changeRequest)
+		})
+
+		t.Run("Running for Pull Requests", func(t *testing.T) {
+			os.Setenv("GITHUB_SHA", testSha)
+			os.Setenv("GITHUB_BASE_REF", "main")
+			os.Setenv("GITHUB_HEAD_REF", testHeadRef)
+			defer restoreGithubFn()
+
+			sha, baseRef, provider, changeRequest := checkGitProvider()
+			assert.Equal(t, testSha, sha)
+			assert.Equal(t, "main", baseRef)
+			assert.Equal(t, "Github", provider)
+			assert.True(t, changeRequest)
+		})
+	})
 
-	assert.Condition(t, func() bool { return keyExists(t, atts, ScmBranch) }, "Branch is not set as scm.branch. Attributes: %v", atts)
+	t.Run("Gitlab", func(t *testing.T) {
+		if originalSha != "" {
+			t.Skip("Tests skipped when running on Github")
```

Comment:
We have to make sure these tests are not ran while developing the project, adding complexity to its maintenance